src/thread-safe-callback.cpp

```diff
@@ -59,5 +60,8 @@ void ThreadSafeCallback::callbackFunc(Napi::Env env,
     }
 
     if (env && callback)
+    {
         callback.Call(context->Value(), args);
+        cleanup();
```

Comment:
I think `cleanup()` should be called anyway, even if `env` or `callback` are not set, otherwise there could be scenarios where cleanup is never done.